app/views/revisions/show.html.erb

```diff
@@ -43,7 +43,7 @@
                     <% if revision == @revision %>
                       Currently Viewing
                     <% elsif revision.event == "update" %>
-                      <%= link_to "View Changes", record_revision_path(@record, revision) %>
+                      <%= link_to "View Changes", record_revision_path(@record, revision, return_to: @return_to_qa ? :qa : nil) %>
```

Comment:
this is another instance of a path helper, but we're solving it in a different way.